hi-ml-histopathology/src/histopathology/preprocessing/create_panda_tiles_dataset.py

```diff
@@ -60,109 +79,114 @@ def save_image(array_chw: np.ndarray, path: Path) -> PIL.Image:
-            dataset_csv_file.write(','.join(CSV_COLUMNS) + '\n')  # write CSV header
+        slide_dir.mkdir(parents=True)
 
-            tiles_failure = 0
-            failed_tiles_csv_path = slide_dir / "failed_tiles.csv"
-            failed_tiles_file = failed_tiles_csv_path.open('w')
-            failed_tiles_file.write('tile_id' + '\n')
+        dataset_csv_path = slide_dir / "dataset.csv"
+        dataset_csv_file = dataset_csv_path.open('w')
+        dataset_csv_file.write(','.join(CSV_COLUMNS) + '\n')  # write CSV header
 
-            logging.info(f"Loading slide {slide_id} ...")
-            loader = LoadPandaROId(WSIReader(), level=level, margin=margin)
+        logging.info(f"Loading slide {slide_id} ...")
+        reader = WSIReader(backend="cucim")
+        loader = LoadPandaROId(reader, level=level, margin=margin)
+        try:
             sample = loader(sample)  # load 'image' and 'mask' from disk
-
-            logging.info(f"Tiling slide {slide_id} ...")
-            image_tiles, mask_tiles, tile_locations, occupancies, _ = \
-                generate_tiles(sample, tile_size, occupancy_threshold)
-            n_tiles = image_tiles.shape[0]
-
-            for i in tqdm(range(n_tiles), f"Tiles ({slide_id[:6]}…)", unit="img", disable=not tile_progress):
-                try:
-                    tile_metadata = save_tile(sample, image_tiles[i], mask_tiles[i], tile_locations[i],
-                                              slide_dir)
-                    tile_metadata['occupancy'] = occupancies[i]
-                    tile_metadata['image'] = os.path.join(slide_dir.name, tile_metadata['image'])
-                    tile_metadata['mask'] = os.path.join(slide_dir.name, tile_metadata['mask'])
-                    dataset_row = ','.join(str(tile_metadata[column]) for column in CSV_COLUMNS)
-                    dataset_csv_file.write(dataset_row + '\n')
-                except Exception as e:
-                    tiles_failure += 1
-                    descriptor = get_tile_descriptor(tile_locations[i]) + '\n'
-                    failed_tiles_file.write(descriptor)
-                    traceback.print_exc()
-                    warnings.warn(f"An error occurred while saving tile "
-                                  f"{get_tile_id(slide_id, tile_locations[i])}: {e}")
-
+        except (ValueError, RuntimeError) as e:
+            logging.error(f'Error loading slide {slide_id}:\n{e}')
             dataset_csv_file.close()
-            failed_tiles_file.close()
-            if tiles_failure > 0:
-                # TODO what we want to do with slides that have some failed tiles?
-                logging.warning(f"{slide_id} is incomplete. {tiles_failure} tiles failed.")
-        except Exception as e:
-            traceback.print_exc()
-            warnings.warn(f"An error occurred while processing slide {slide_id}: {e}")
+            shutil.rmtree(slide_dir)
+            return
+        logging.info(f"Tiling slide {slide_id} ...")
```

Comment:
The following block was in the try/except before. In the current version we assume the error can come only from loading the slide. If I remember correctly, we had errors also on saving. It might be safer to wrap also this block in a try/except.

Reply:
I got no errors on saving. They might have come from the issue with `squeeze`. I have narrowed down the try/except blocks to identify why errors happen and properly decide what to do with them.

Reply:
as long as you can create the full dataset without errors looks OK to me - separate try/except would have been OK to capture different errors

Reply:
I've updated the code to make error catching more granular. There are 4 slides with an empty mask and around 100 slides without an associated mask. All those slides are currently being ignored. For these cases, we could resort to computing the mask ourselves, but that might be out of the scope of this PR.